### PR TITLE
round position updates to 2 decimal places

### LINKED ISSUE #61: Position should be rounded to 2 decimal places to avoid floating point precision errors in collision fixes

### PR DESCRIPTION
**Pull request by tlein**
Resolves #61

### CODE REVIEW

File: Engine/Src/Ancona/Core2D/Systems/Position/PositionSystem.cpp [hunk 1]
```diff
@@ -51,9 +51,25 @@ bool PositionComponent::onGround() const
...
 void PositionComponent::Update(float delta) {
-    _position += delta * _velocity;
+    if (_velocity.x != 0.0f || _velocity.y != 0.0f) 
+    {
+        _actualPosition += delta * _velocity;
+        RoundPosition();
+    }
 }
+
+void PositionComponent::RoundPosition()
```
Comment (tlein) [inline Engine/Src/Ancona/Core2D/Systems/Position/PositionSystem.cpp:71]: We should round the position in the setter as well

Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/Position/PositionSystem.cpp:71]: fixed

File: Engine/Src/Ancona/Core2D/Systems/Position/PositionSystem.cpp [hunk 2]
```diff
@@ -51,9 +51,32 @@ bool PositionComponent::onGround() const
...
+    if (_velocity.x != 0.0f || _velocity.y != 0.0f) 
+    {
+        _actualPosition += delta * _velocity;
+        RoundPosition();
+    }
 }
+
+void PositionComponent::RoundPosition() 
+{
+    _position.x = roundf(_actualPosition.x * 100) / 100;
```
Comment (jeffswenson) [inline Engine/Src/Ancona/Core2D/Systems/Position/PositionSystem.cpp]: You should use 128 instead of 100. It will be slightly better behaved.

Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/Position/PositionSystem.cpp]: fixed

**Pull request merged**